### PR TITLE
Update Sunflower from M2 to M3 (#852)

### PR DESCRIPTION
**Pull request by arriolac**
Resolves #852. A majority of the migration has been done by @MagicalMeghan. The remaining work left is to update colors to match the material3 color spec.

### CODE REVIEW

**Review by JolandaVerhoef — APPROVED** [review]

File: app/src/main/java/com/google/samples/apps/sunflower/compose/plantlist/PlantListItemView.kt [hunk 1]
```diff
@@ -48,13 +48,13 @@ fun PhotoListItem(photo: UnsplashPhoto, onClick: () -> Unit) {
...
 
-@OptIn(ExperimentalMaterialApi::class)
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ImageListItem(name: String, imageUrl: String, onClick: () -> Unit) {
     Card(
         onClick = onClick,
-        elevation = dimensionResource(id = R.dimen.card_elevation),
-        shape = MaterialTheme.shapes.card,
+        shape = MaterialTheme.shapes.medium,
```
Comment (Gurupreet) [inline app/src/main/java/com/google/samples/apps/sunflower/compose/plantlist/PlantListItemView.kt:56]: I think default shape should be medium itself.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!--
```
Comment (Gurupreet) [inline app/src/main/res/values/styles.xml:1]: Is this file still required?

**Review by Gurupreet — APPROVED** [review]

**Review by florina-muntenescu — COMMENTED** [review]
Update the readme with latest screenshots as well.

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (MagicalMeghan) [reply app/src/main/res/values/styles.xml:1]: it is per the discussion in the best practices doc - but it should be renamed to themes.xml

File: app/src/main/java/com/google/samples/apps/sunflower/ui/Theme.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            // dark icons if we're in light theme
+            systemUiController.setSystemBarsColor(
+                color = Color.Transparent,
+                darkIcons = useDarkIcons
+            )
+            onDispose {}
+        }
+    }
+
+    MaterialTheme(
```
[the remark this answers is not in the document]
Comment (MagicalMeghan) [reply app/src/main/java/com/google/samples/apps/sunflower/ui/Theme.kt:135]: added
